src/components/ModalWrapper.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react'
+const { bool, func, object, string } = React.PropTypes
+import BrowseTopicsModal from './BrowseTopicsModal'
+import ShareTopicModal from '../containers/ShareTopicModal'
+import ExpandedPostModal from '../containers/ExpandedPostModal'
+import DirectMessageModal from '../containers/DirectMessageModal'
+import ChecklistModal from '../containers/ChecklistModal'
+import TagEditorModal from '../containers/TagEditorModal'
+import AddLogoModal from '../containers/AddLogoModal'
+import InviteModal from '../containers/InviteModal'
+import PostEditorModal from '../containers/PostEditorModal'
+import ImageModal from '../containers/ImageModal'
+import { NotificationsModal } from '../containers/Notifications'
+import { ThreadsModal } from '../containers/ThreadsDropdown'
+import { closeModal } from '../actions'
+import { get } from 'lodash/fp'
+
+export class ModalWrapper extends React.Component {
+  static propTypes = {type: string, params: object, top: bool, bottom: bool}
+  static defaultProps = {top: true}
+  static contextTypes = {dispatch: func}
+
+  lockScrolling = () => {
+    window.scrollTo(0, this.lockedScrollTop)
+  }
+
+  componentDidMount () {
+    if (this.props.bottom) {
+      this.lockedScrollTop = document.body.scrollTop
+      window.addEventListener('scroll', this.lockScrolling)
+    }
+  }
+
+  componentWillUnmount () {
+    if (this.props.bottom) window.removeEventListener('scroll', this.lockScrolling)
+  }
+
+  render () {
+    const { type, params, top } = this.props
+    const { dispatch } = this.context
+    if (!type) return null
+
+    let modal, clickToClose
+    const close = () => dispatch(closeModal())
```

Comment:
Connect this component with this in mapDispatchToProps, bindActionCreators:

`closeClickHandler: () => closeModal()`